staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -52,14 +53,15 @@ type tlsCacheKey struct {
 	nextProtos         string
 	dial               string
 	disableCompression bool
+	proxyURL           *url.URL
 }
 
 func (t tlsCacheKey) String() string {
 	keyText := "<none>"
 	if len(t.keyData) > 0 {
 		keyText = "<redacted>"
 	}
-	return fmt.Sprintf("insecure:%v, caData:%#v, certData:%#v, keyData:%s, getCert: %s, serverName:%s, dial:%s disableCompression:%t", t.insecure, t.caData, t.certData, keyText, t.getCert, t.serverName, t.dial, t.disableCompression)
+	return fmt.Sprintf("insecure:%v, caData:%#v, certData:%#v, keyData:%s, getCert: %s, serverName:%s, dial:%s disableCompression:%t, proxyURL: %q", t.insecure, t.caData, t.certData, keyText, t.getCert, t.serverName, t.dial, t.disableCompression, t.proxyURL)
```

Comment:
none of the other data in the key was secret material, but according to earlier comments the proxy url could contain a password... should we avoid making it part of the cache key as-is?

Reply:
proxy URL should probably be in the cache since it's part of the cluster object and transport. I could hash it?

Reply:
Fixed by modeling the proxy func like we do with Dial